testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2064,6 +2064,28 @@ public void testHandlingPartitionSchemaEvolutionInPartitionMetadata()
                 ImmutableMap.of("old_partition_key", "3", "new_partition_key", "null", "value_day", "null", "value_month", "null")));
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testAlterTableExecuteProceduresOnEmptyTable()
```

Comment:
nit: I would move to after `testOptimizeOnV2IcebergTable` or somewhere. The above `testHandlingPartitionSchemaEvolutionInPartitionMetadata` test is the only usage of following `validatePartitioning` helper method.